sandbox/modules/identity/src/main/java/org/opensearch/identity/authz/OpenSearchPermission.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.authz;
+
+import org.apache.shiro.authz.Permission;
+import org.apache.shiro.authz.permission.WildcardPermission;
+import org.opensearch.common.Glob;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class OpenSearchPermission implements Permission {
+
+    private String actionName;
+
+    private List<String> indexPatterns;
+
+    public OpenSearchPermission(String permission) {
+        Objects.requireNonNull(permission);
+        String[] permissionParts = permission.split("\\|");
+        // turns an action like indices:data/read/search into indices:data:read:search to leverage shiro's WildcardPermission
+        this.actionName = permissionParts[0].replace("/", ":");
+        if (permissionParts.length > 1) {
+            this.indexPatterns = Arrays.stream(permissionParts[1].split(",")).map(String::trim).collect(Collectors.toList());
+        }
+    }
+
+    @Override
+    public boolean implies(Permission p) {
+        if (!(p instanceof OpenSearchPermission)) {
+            return false;
+        }
+        OpenSearchPermission requestedPermission = (OpenSearchPermission) p;
+
+        WildcardPermission wp = new WildcardPermission(this.actionName);
+        WildcardPermission wp2 = new WildcardPermission(requestedPermission.actionName);
+        if (!wp.implies(wp2)) {
+            return false;
+        }
+        if (this.indexPatterns == null || this.indexPatterns.isEmpty()) {
+            // TODO Find a better way to do this, this means only the permission does not require a resource
+            // and is permitted by name alone
+            return true;
+        }
+        if (requestedPermission.indexPatterns == null || requestedPermission.indexPatterns.isEmpty()) {
+            return false;
+        }
+
+        // Uncomment the following lines if index name -> concrete index resolution is required
```

Comment:
To fully uncomment this block and use the `IndexNameExpressionResolver` then the following needs to be added to `IdentityPlugin`:

```
@Override
public Collection<Class<? extends LifecycleComponent>> getGuiceServiceClasses() {

    if (!enabled) {
        return Collections.emptyList();
    }

    final List<Class<? extends LifecycleComponent>> services = new ArrayList<>(1);
    services.add(GuiceHolder.class);
    return services;
}

public static class GuiceHolder implements LifecycleComponent {

    private static ClusterService clusterService;
    private static RepositoriesService repositoriesService;

    @Inject
    public GuiceHolder(final RepositoriesService repositoriesService, final ClusterService clusterService) {
        GuiceHolder.repositoriesService = repositoriesService;
        GuiceHolder.clusterService = clusterService;
    }

    public static RepositoriesService getRepositoriesService() {
        return repositoriesService;
    }

    public static ClusterService getClusterService() {
        return clusterService;
    }

    @Override
    public void close() {}

    @Override
    public Lifecycle.State lifecycleState() {
        return null;
    }

    @Override
    public void addLifecycleListener(LifecycleListener listener) {}

    @Override
    public void removeLifecycleListener(LifecycleListener listener) {}

    @Override
    public void start() {}

    @Override
    public void stop() {}

}
```

Reply:
I don't think we should be attempting to resolve the name ever, if you have permissions to the URL you get access to target, so if there is an alias, so long as you have permission, its allowed.  This prevents 'transitive' permissions due to the creation of aliases or other systems... views?, food food thought.